### PR TITLE
V6

### PR DESCRIPTION
**Pull request by tabaktoni**
## Motivation and Resolution
Provider / Provider Responses / Response parser
Fixed, Cleaned, Guard

## Checklist:

- [ ] Performed a self-review of the code
- [ ] Rebased to the last commit of the target branch (or merged it into my branch)
- [ ] Linked the issues which this PR resolves
- [ ] Documented the changes in code (API docs will be generated automatically)
- [ ] Updated the tests
- [ ] All tests are passing

### CODE REVIEW

File: src/utils/responseParser/rpc.ts [hunk 1]
```diff
@@ -31,67 +23,66 @@ export class RPCResponseParser
...
-      timestamp: res.timestamp,
-      block_hash: 'block_hash' in res ? res.block_hash : '',
-      block_number: 'block_number' in res ? res.block_number : -1,
-      new_root: 'new_root' in res ? res.new_root : '',
-      parent_hash: res.parent_hash,
-      status: 'status' in res ? (res.status as BlockStatus) : BlockStatus.PENDING,
-      transactions: res.transactions,
-    };
+  public parseGetBlockResponse(res: RPC.BlockWithTxHashes): GetBlockResponse {
+    return 'status' in res ? res : { ...res, status: 'PENDING' };
```
Comment (penovicp) [inline src/utils/responseParser/rpc.ts:31]:
```suggestion
    return { status: 'PENDING', ...res };
```

File: src/provider/rpc.ts [hunk 2]
```diff
@@ -339,9 +347,8 @@ export class RpcProvider implements ProviderInterface {
   }
 
   public async callContract(call: Call, blockIdentifier?: BlockIdentifier) {
-    return this.channel
-      .callContract(call, blockIdentifier)
-      .then(this.responseParser.parseCallContractResponse);
+    return this.channel.callContract(call, blockIdentifier);
+    // .then(this.responseParser.parseCallContractResponse);
```
Comment (penovicp) [inline src/provider/rpc.ts:351]: Straggler?

File: src/types/api/rpcspec_0_6/components.ts [hunk 3]
```diff
@@ -134,6 +134,7 @@ export type STATE_DIFF = {
 export type PENDING_STATE_UPDATE = {
   old_root: FELT;
   state_diff: STATE_DIFF;
+  block_hash: never; // diverge: this make it distinct
```
Comment (penovicp) [inline src/types/api/rpcspec_0_6/components.ts:137]:
```suggestion
  block_hash: never; // diverge: this makes it distinct
```

File: src/utils/provider.ts [hunk 4]
```diff
@@ -163,22 +166,11 @@ export class Block {
 
   toString = () => this.hash;
 
-  get sequencerIdentifier(): SequencerIdentifier {
+  /*   get sequencerIdentifier(): SequencerIdentifier {
```
Comment (penovicp) [inline src/utils/provider.ts:169]: Straggler?

File: src/utils/provider.ts [hunk 5]
```diff
@@ -192,3 +184,29 @@ export function isVersion(version: '0.5' | '0.6', response: string) {
...
+ */
+export function isPendingTransaction(
+  response: GetTransactionReceiptResponse
+): response is RPC.PendingReceipt {
+  return !('block_hash' in response);
+}
+
+/**
+ * Guard Pending State Update
+ * ex. if(isPendingStateUpdate(stateUpdate)) throe Error('Update must be final')
```
Comment (penovicp) [inline src/utils/provider.ts:206]:
```suggestion
 * ex. if(isPendingStateUpdate(stateUpdate)) throw Error('Update must be final')
```

**Review by penovicp — APPROVED** [review]
Left some minor comments, lgtm

File: src/utils/provider.ts (hunk 4 above)
Comment (tabaktoni) [reply src/utils/provider.ts:169]: Intentional

**Pull request merged**